notify.go

```diff
@@ -18,25 +18,10 @@ func (as *AmbientAutoNAT) OpenedStream(net network.Network, s network.Stream) {}
 func (as *AmbientAutoNAT) ClosedStream(net network.Network, s network.Stream) {}
 
 func (as *AmbientAutoNAT) Connected(net network.Network, c network.Conn) {
-	p := c.RemotePeer()
-
-	go func() {
-		// add some delay for identify
-		time.Sleep(AutoNATIdentifyDelay)
-
-		protos, err := as.host.Peerstore().SupportsProtocols(p, AutoNATProto)
-		if err != nil {
-			log.Debugf("error retrieving supported protocols for peer %s: %s", p, err)
-			return
-		}
-
-		if len(protos) > 0 {
-			log.Infof("Discovered AutoNAT peer %s", p.Pretty())
-			as.mx.Lock()
-			as.peers[p] = as.host.Peerstore().Addrs(p)
-			as.mx.Unlock()
-		}
-	}()
+	select {
+	case as.candidatePeers <- c:
+	default:
```

Comment:
One of the motivations for this change was that it seemed suboptimal to have yet another cache of peers that could get out of sync. If our expectation is that most DHT peers will be running autonat services, we should be able to choose from among them and not have another cache.

Reply:
I see. Do we plan to enable AutoNAT by default on all DHT peers ?

Reply:
I think that's the thought: The autonat service will be active when the node believes it is publicly reachable, so the same condition as participating as a DHT server

Reply:
Based on an offline discussion with @willscott  & @Stebalien , we've agreed that running AutoNAT client & service will be the default behaviour of a libp2p peer & users will have to opt out if they want to. This will  likely ensure that our connected peers will have some AutoNAT servers & so we don't need to cache server addresses anymore.